skimage/metrics/overlap.py

```diff
@@ -0,0 +1,221 @@
+"""
+Module defining a number of functions to quantify the overlap between shapes.
+for instance rectangles representing detections by bounding-boxes.
+
+"""
+
+class Rectangle():
+    """
+    Represents rectangles which position corresponds to the (r,c) coordinates for the top left corner.
+    Rectangle objects have a number of attributes:
+        - coordinates for the bottom right corner
+        - rectangle dimensions
+        - area
+    as well as utility functions :
+        - computation of the intersection of 2 rectangles
+        - equivalent area for te union of 2 rectangles.
+    """
+
+    def __init__(self, top_left, bottom_right=None, size=None):
```

Comment:
```suggestion
    def __init__(self, top_left, *, bottom_right=None, size=None):
```

In order to avoid confusion, I would force the arguments after `top_left` to be keyword-only. That way users will always have to document `bottom_right` or `size`.